### PR TITLE
Homework #6

### PR DESCRIPTION
**Pull request by AndryNosach**
Done:
1. Saving all quiz structure from web forms into db.
2. Deleting quiz from db.
3. User registration and authorization using db, password is encrypted.

SQL creating schema scripte file and  model mysql workbench file are in the root.

### CODE REVIEW

File: Nosach/web-quiz-project/src/main/java/service/QuestionService.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        return qd.addQuestion(question, quizId);
+    }
+
+    public Question getQuestion(int id) {
+        QuestionDAO qd = new QuestionDAOImpl();
+        return  qd.getQuestion(id);
+    }
+
+    public List<Question> getQuestionsForQuiz(int quizId) {
+        QuestionDAO qd = new QuestionDAOImpl();
```
Comment (dinamio) [inline Nosach/web-quiz-project/src/main/java/service/QuestionService.java]: Better create one dao per service

File: Nosach/web-quiz-project/src/main/java/servlet/RegisterServlet.java [hunk 2]
```diff
@@ -21,15 +25,16 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws S
...
         String email = req.getParameter("email");
+        String name = req.getParameter("name");
 
-        try (FileWriter fw = new FileWriter(getServletContext().getRealPath("/resources/properties/users.properties"), true)) {
 
-            fw.write(login + "#" + pass + "#" + email + "\n");
-            fw.flush();
-        }
+        UserService us = new UserService();
+        PasswordEncoder encoder = new BCryptPasswordEncoder();
```
Comment (dinamio) [inline Nosach/web-quiz-project/src/main/java/servlet/RegisterServlet.java]: Consider to pass encoding logic to service, servlet is not right place for this, on my point of view

**Review by dinamio — APPROVED** [review]
Good. Please, check comments, and merge after you fix them.

File: Nosach/web-quiz-project/src/main/java/service/QuestionService.java (hunk 1 above)
Comment (AndryNosach) [reply Nosach/web-quiz-project/src/main/java/service/QuestionService.java]: fixed

File: Nosach/web-quiz-project/src/main/java/servlet/RegisterServlet.java (hunk 2 above)
Comment (AndryNosach) [reply Nosach/web-quiz-project/src/main/java/servlet/RegisterServlet.java]: fixed

**Pull request merged**